nodes/dope

```diff
@@ -168,12 +170,21 @@ class DopeNode(object):
     def image_callback(self, image_msg, camera_info):
         """Image callback"""
 
-        # Update camera matrix
-        P = np.matrix(camera_info.P, dtype='float64')
-        P.resize((3, 4))
-        camera_matrix = P[:, :3]
+        # Update camera matrix and distortion coefficients
+        if self.input_is_rectified:
+            P = np.matrix(camera_info.P, dtype='float64')
+            P.resize((3, 4))
+            camera_matrix = P[:, :3]
+            dist_coeffs = np.zeros((4, 1))
+        else:
+            camera_matrix = np.matrix(camera_info.K, dtype='float64')
+            camera_matrix.resize((3, 3))
+            dist_coeffs = np.matrix(camera_info.D, dtype='float64')
+            dist_coeffs.resize((len(camera_info.D), 1))
+
         for m in self.models:
             self.pnp_solvers[m].set_camera_intrinsic_matrix(camera_matrix)
+            self.pnp_solvers[m].set_dist_coeffs(dist_coeffs)
```

Comment:
These changes are all correct but I think when we run DOPE on a non-rectified image it won't be as accurate as with a rectified one? I think a better workflow should be rectify the input image if the camera is non-rectified. It could be a feature for the future.

Reply:
Thanks for your comment! I intentionally left this out, because in ROS the standard way would be to use this: http://wiki.ros.org/image_proc. The user should start a image_proc/rectify nodelet separately and feed the rectified topic to DOPE.